### ISSUE TITLE
Problem synchronizing LDAP (AD) users

### ISSUE THREAD

**Issue by daften** [snipe/snipe-it#2036]
#### Expected Behavior (or desired behavior if a feature request)

The users to be imported

---
#### Actual Behavior

We set the filter query to a specific CN so only a limited set of users would be imported (as a trial). When we run the import, we get the following result in the front-end:
![screen shot 2016-05-12 at 14 57 30](https://cloud.githubusercontent.com/assets/1221024/15215264/e4fd835c-1851-11e6-9618-d8d71c2ffca9.png)

If we look at the $results array by outputting it to log, it doesn't seem to be in the expected format:

```
(
    [count] => 1
    [0] => Array
        (
            [objectclass] => Array
                (
                    [count] => 2
                    [0] => top
                    [1] => group
                )

            [0] => objectclass
            [cn] => Array
                (
                    [count] => 1
                    [0] => .........
                )

            [1] => cn
            [description] => Array
                (
                    [count] => 1
                    [0] => .........
                )

            [2] => description
            [member] => Array
                (
                    [count] => 38
                    [0] => CN=.........,OU=.........,OU=.........,OU=.........,DC=.........,DC=.........,DC=.........
                    [1] => ......... until [37]
                )

            [3] => member
            [distinguishedname] => Array
                (
                    [count] => 1
                    [0] => CN=.........,OU=.........,OU=.........,OU=.........,DC=.........,DC=.........,DC=.........
                )

            [4] => distinguishedname
            [instancetype] => Array
                (
                    [count] => 1
                    [0] => 4
                )

            [5] => instancetype
            [whencreated] => Array
                (
                    [count] => 1
                    [0] => 20100309092114.0Z
                )

            [6] => whencreated
            [whenchanged] => Array
                (
                    [count] => 1
                    [0] => 20160412115758.0Z
                )

            [7] => whenchanged
            [usncreated] => Array
                (
                    [count] => 1
                    [0] => 133304
                )

            [8] => usncreated
            [memberof] => Array
                (
                    [count] => 18
                    [0] => CN=.........,OU=.........,OU=.........,OU=.........,OU=......... Gent,DC=.........,DC=.........,DC=.........
                    [1] => ......... until [17]
                )

            [9] => memberof
            [usnchanged] => Array
                (
                    [count] => 1
                    [0] => 214386207
                )

            [10] => usnchanged
            [name] => Array
                (
                    [count] => 1
                    [0] => .........
                )

            [11] => name
            [objectguid] => Array
                (
                    [count] => 1
                    [0] => ��h�\��M���]���
                )

            [12] => objectguid
            [objectsid] => Array
                (
                    [count] => 1
                    [0] => /:
_8`:B_�
                )

            [13] => objectsid
            [samaccountname] => Array
                (
                    [count] => 1
                    [0] => .........
                )

            [14] => samaccountname
            [samaccounttype] => Array
                (
                    [count] => 1
                    [0] => 268435456
                )

            [15] => samaccounttype
            [managedby] => Array
                (
                    [count] => 1
                    [0] => CN=.........,OU=.........,OU=.........,DC=.........,DC=.........,DC=.........
                )

            [16] => managedby
            [grouptype] => Array
                (
                    [count] => 1
                    [0] => -2147483646
                )

            [17] => grouptype
            [objectcategory] => Array
                (
                    [count] => 1
                    [0] => CN=.........,CN=.........,CN=.........,DC=.........,DC=.........,DC=.........
                )

            [18] => objectcategory
            [dscorepropagationdata] => Array
                (
                    [count] => 5
                    [0] => 20160422125235.0Z
                    [1] => 20160415093259.0Z
                    [2] => 20160415093123.0Z
                    [3] => 20160415090429.0Z
                    [4] => 16010714223649.0Z
                )

            [19] => dscorepropagationdata
            [count] => 20
            [dn] => CN=.........,OU=.........,OU=.........,OU=.........,DC=.........,DC=.........,DC=.........
        )
)
```

I replaced the private data with ......... for security reasons

The filter used is cn=<cn_name>

---
#### Please confirm you have done the following before posting your bug report:
- [ X] I have enabled debug mode
- [ X] I have read [checked the Common Issues page](http://docs.snipeitapp.com/common-issues.html)

---
#### Please provide answers to these questions before posting your bug report:
- Version of Snipe-IT you're running: 2.1.0
- What OS and web server you're running Snipe-IT on: Debian wheezy with apache 2.2 and php-fpm 5.6
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc): manual installation
- If you're getting an error in your browser, include that error: /
- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error: /admin/users/ldap
- If a stacktrace is provided in the error, include that too.: no
- Any errors that appear in your browser's error console.: /
- Confirm whether the error is [reproduceable on the demo](https://snipeitapp.com/demo). /
- Include any additional information you can find in `app/storage/logs` and your webserver's logs. /
- Include what you've done so far in the installation, and if you got any error messages along the way. /
- Indicate whether or not you've manually edited any data directly in the database /

**Comment by ddreier** [comment]
Can you post a properly-redacted screenshot of your LDAP settings?

**Comment by daften** [comment]
Hi,

Sure:
![screen shot 2016-05-13 at 10 48 59](https://cloud.githubusercontent.com/assets/1221024/15242837/93590be6-18f8-11e6-9e63-191294495514.png)
We use email as username in this application (it was used before AD was inserted).

**Comment by ddreier** [comment]
Not really. The LDAP settings look correct (except for the Active Flag, which `active` is not in AD by default), but the result array @daften posted is missing a bunch of fields (including `sn` and `givenname`).

@daften have you tried a different filter? (`&(objectcategory=person)(objectclass=user)` is a decent one to try)

**Comment by daften** [comment]
Thanks for the feedback, I will try a different filter, but we need to restrict to a small group. Our AD has a couple of thousand users in it, so not a very good idea to start a sync without a decent filter

**Comment by daften** [comment]
We've tested it and the hint by @ddreier works, we also had to use memberOf=cn=... as the last part of the filter. Using something like memberOf:1.2.840.113556.1.4.1941:= 
cn=Test,ou=East,dc=Domain,dc=com worked perfectly (searches hierarchically).
Our original filter query just returned the group itself.

Is this something that should be added in the interface or documentation somewhere? If so, where preferably? I can create a PR for this later this month then.
